sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/query/SingleGroupAggregator.java

```diff
@@ -0,0 +1,316 @@
+        }
+
+        SingleGroupAggregator singleGroupAggregator;
+
+        if (hasSelectValue) {
+            if (!aggregates.isEmpty()) {
+                // SELECT VALUE <AGGREGATE>
+                singleGroupAggregator = SelectValueAggregateValues.create(
+                    aggregates.get(0),
+                    continuationToken);
+            } else {
+                // SELECT VALUE <NON AGGREGATE>
+                singleGroupAggregator = SelectValueAggregateValues.create(
+                    null, continuationToken);
+            }
+        } else {
+            singleGroupAggregator = SelectListAggregateValues.create(
+                aggregateAliasToAggregateType,
+                orderedAliases,
+                continuationToken);
+        }
+
+        return singleGroupAggregator;
+    }
+
+    /**
+     * Adds the payload for group by values
+     * @param values
+     */
+    public abstract void addValues(Document values);
+
+    /**
+     * Forms the final result of the grouping.
+     * @return Document
+     */
+    public abstract Document getResult();
+
+    public abstract Resource getDocumentContinuationToken();
+
+    /**
+     * For SELECT VALUE queries there is only one value for each grouping.
+     * This class just helps maintain that and captures the first value across all continuations.
+     */
+    public static class SelectValueAggregateValues extends SingleGroupAggregator {
+        private final AggregateValue aggregateValue;
+
+        public SelectValueAggregateValues(AggregateValue aggregateValue) {
+            //TODO: null check
+            this.aggregateValue = aggregateValue;
+        }
+
+        public static SingleGroupAggregator create(AggregateOperator aggregateOperator, String continuationToken) {
+            AggregateValue aggregateValue = AggregateValue.create(aggregateOperator, continuationToken);
+            return new SelectValueAggregateValues(aggregateValue);
+        }
+
+        @Override
+        public void addValues(Document values) {
+            this.aggregateValue.addValue(values);
+        }
+
+        @Override
+        public Document getResult() {
+            Document document;
+            Object result = aggregateValue.getResult();
+            if (result instanceof Document) {
+                document = (Document) aggregateValue.getResult();
+            } else {
+                document = new Document();
+                if (result instanceof Undefined) {
+                    result = null;
+                }
+                // This is for value queries.Need to append value property for scalar values (non key value pairs)
+                // to make them a key value pair document as our impl is based on Document.
+                document.set(Constants.Properties.VALUE, result);
+            }
+            return document;
+        }
+
+        @Override
+        public Resource getDocumentContinuationToken() {
+            return null;
```

Comment:
why continuation token is null here?

Reply:
Continuation token is not yet supported for this, this method not referenced yet. Will change this when implementing cont.token.